src/rt/tracegc.d

```diff
@@ -134,9 +156,16 @@ private string generateWrapper(alias Declaration)()
     auto name = __traits(identifier, Declaration);
     auto param_idx = findParamIndex(type_string);
 
-    auto new_declaration = type_string[0 .. param_idx] ~ " " ~ name
-        ~ "Trace(string file, int line, string funcname, "
-        ~ type_string[param_idx+1 .. $];
+    static if (pos == ParamPos.front)
+        auto new_declaration = type_string[0 .. param_idx] ~ " " ~ name
+            ~ "Trace(string file, int line, string funcname, "
+            ~ type_string[param_idx+1 .. $];
+    else static if (pos == ParamPos.back)
+        auto new_declaration = type_string[0 .. param_idx] ~ " " ~ name
+            ~ "Trace(" ~ type_string[param_idx+1 .. $-1]
+            ~ `, string file = "", int line = 0, string funcname = "")`;
```

Comment:
When appending parameters to the end, we have to deal with the possibility of default parameters. Since we can't add non-default parameters after default ones, we need to make file, line, funcname default parameters as well. This shouldn't be a problem, since the declarations in `core.memory.GC` should already insert concrete values for these parameters on the caller's end, so these default values shouldn't ever happen (unless user code calls `gc_xxxTrace` directly -- but that's beyond our ability to control).